elm2nix-0.1.0 build failure with http-client-0.6

Use `defaultRequest` instead of `def`

```
[2 of 4] Compiling Elm2Nix.PackagesSnapshot ( src/Elm2Nix/PackagesSnapshot.hs, dist/build/Elm2Nix/PackagesSnapshot.o )

src/Elm2Nix/PackagesSnapshot.hs:97:19: error:
    • No instance for (data-default-class-0.1.2.0:Data.Default.Class.Default
                         Req.HttpConfig)
        arising from a use of ‘def’
    • In the first argument of ‘Req.runReq’, namely ‘def’
      In the expression: Req.runReq def
      In a stmt of a 'do' block:
        r <- Req.runReq def
               $ Req.req
                   Req.POST
                   (Req.https "package.elm-lang.org" Req./: "all-packages")
                   Req.NoReqBody
                   Req.jsonResponse
                   mempty
   |
97 |   r <- Req.runReq def $
   |                   ^^^
```

I was able to reproduce this locally like so:

```bash
stack unpack elm2nix-0.1.0 && cd elm2nix-0.1.0
edit stack.yaml # add the following stack.yaml
stack build --test --bench --no-run-benchmarks --fast
```

```yaml
# stack.yaml
resolver: nightly-2019-02-19
extra-deps:
- here-1.2.13
- haskell-src-meta-0.8.1

```

--- a/ChangeLog.md
+++ b/ChangeLog.md
@@ -1,5 +1,11 @@
 # Changelog for elm2nix
 
+## 0.1.1 (2019-02-19)
+
+### Changes
+
+- #22 Support req-0.1.0.0 (@domenkozar)
+
 ## 0.1.0 (2018-12-28)
 
 - Initial release (@domenkozar)

--- a/elm2nix.cabal
+++ b/elm2nix.cabal
@@ -3,7 +3,7 @@
 -- see: https://github.com/sol/hpack
 
 name:           elm2nix
-version:        0.1.0
+version:        0.1.1
 synopsis:       Turn your Elm project into buildable Nix project
 description:    Please see the README on Github at <https://github.com/domenkozar/elm2nix#readme>
 homepage:       https://github.com/domenkozar/elm2nix#readme

--- a/src/Elm2Nix/PackagesSnapshot.hs
+++ b/src/Elm2Nix/PackagesSnapshot.hs
@@ -8,6 +8,7 @@
 
 -}
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE CPP #-}
 module Elm2Nix.PackagesSnapshot
   ( snapshot
   ) where
@@ -17,7 +18,10 @@ import qualified Data.Aeson as Aeson
 import qualified Data.Binary as Binary
 import Data.Binary (Binary, put, get, putWord8, getWord8)
 import qualified Data.Map as Map
+#if MIN_VERSION_req(2,0,0)
+#else
 import Data.Default (def)
+#endif
 import Data.Map (Map)
 import Data.Text (Text)
 import qualified Data.Text as Text
@@ -92,9 +96,15 @@ instance Binary PackageRegistry where
   get = liftM2 PackageRegistry get get
   put (PackageRegistry a b) = put a >> put b
 
+#if MIN_VERSION_req(2,0,0)
+defHttpConfig = Req.defaultHttpConfig
+#else
+defHttpConfig = def
+#endif
+
 snapshot :: String -> IO ()
 snapshot dir = do
-  r <- Req.runReq def $
+  r <- Req.runReq defHttpConfig $
     Req.req
     Req.POST
     (Req.https "package.elm-lang.org" Req./: "all-packages")

